### PR TITLE
fix!: Fix on QuorumDataRequests and refactoring

### PR DESCRIPTION
**Pull request by ogabrielides**
This PR includes:

- Fix on **important** typo errors introduced on #4935 [here](https://github.com/dashpay/dash/pull/4935/files#diff-0511b9a66cb825475c3a80b6666371bd6d9f3d7594d83657e748d0a243b99cd9R875) and [here](https://github.com/dashpay/dash/pull/4935/files#diff-0511b9a66cb825475c3a80b6666371bd6d9f3d7594d83657e748d0a243b99cd9R903)

- Refactoring improving code readability. Therefore, I still don't understand the purpose of the boolean variable. To be renamed later on a new PR once we figure out.

### CODE REVIEW

**Review by PastaPastaPasta — CHANGES_REQUESTED** [review]
CI fails

File: src/llmq/quorums.cpp [hunk 1]
```diff
@@ -616,10 +616,11 @@ void CQuorumManager::ProcessMessage(CNode* pFrom, const std::string& msg_type, C
         };
 
         {
-            LOCK2(cs_main, cs_data_requests);
-            auto quorumHash = request.GetQuorumHash();
-            auto llmqType = (uint8_t) request.GetLLMQType();
-            auto key = std::make_tuple(pFrom->GetVerifiedProRegTxHash(), false, quorumHash, llmqType);
+            CQuorumDataRequestKey key;
```
Comment (UdjinM6) [inline src/llmq/quorums.cpp:619]:
```suggestion
            LOCK2(cs_main, cs_data_requests);
            CQuorumDataRequestKey key;
```

**Review by UdjinM6 — CHANGES_REQUESTED** [review]

File: src/llmq/quorums.cpp (hunk 1 above)
Comment (ogabrielides) [reply src/llmq/quorums.cpp:620]: @UdjinM6 Oups. Good catch :)

**Review by PastaPastaPasta — APPROVED** [review]
utACK for squash merge

**Review by UdjinM6 — APPROVED** [review]
utACK

**Comment by ogabrielides** [comment]
@UdjinM6 @PastaPastaPasta merge?

**Pull request merged**